config/build/rollup.config.js

```diff
@@ -44,62 +28,20 @@ const makePackageBasedPath = (pkgName, pkgRelPath) => {
         path.join(rootDir, "packages", pkgName, "package.json"),
     );
     const pkgJson = require(pkgPath);
-    return path.normalize(path.join("packages", pkgName, pkgJson.source));
-};
-
-/**
- * Generate the rollup output configuration for a given package
- */
-const createOutputConfig = (pkgName, format, targetFile) => ({
-    file: makePackageBasedPath(pkgName, targetFile),
-    sourcemap: true,
-    format,
-
-    // These two settings are to keep the builds as similar to pre-Rollup v4 as
-    // possible until we get rid of CJS builds.
-    // See: https://rollupjs.org/migration/#changed-defaults
-    esModule: true,
-    interop: "compat",
-
-    // Governs names of CSS files (for assets from CSS use `hash` option for
-    // url handler).
-    // Note: using value below will put `.css` files near js,
-    // but make sure to adjust `hash`, `assetDir` and `publicPath`
-    // options for url handler accordingly.
-    assetFileNames: "[name][extname]",
-});
-
-/**
- * Get a set of strings from a given string, returning the defaults
- *
- * This assumes comma-delimited strings.
- */
-const getSetFromDelimitedString = (arg, defaults) => {
-    const values =
-        arg != null && arg.length > 0
-            ? arg
-                  .split(",")
-                  .map((p) => p.trim())
-                  .filter(Boolean)
-            : [];
-    return new Set(values.length ? values : defaults);
+    return path.normalize(
+        path.join("packages", pkgName, pkgJson.exports["."].source),
```

Comment:
Not sure I get what's happening here with `.exports["."]`

Reply:
Ah, I understand now. Each package.json will be in the shape of:

```
{
    "exports": {
        ".": {
            "import": "./dist/index.js",
            "types": "./dist/index.d.ts",
            "source": "./src/index.ts"
        }
    }
}
```

> When using the ["exports"](https://nodejs.org/api/packages.html#exports) field, custom subpaths can be defined along with the main entry point by treating the main entry point as the "." subpath

So I guess we don't get a say in what we call `"."`.

Reply:
Correct.  There's a simpler form of the "exports" field like this: `"exports": "./dist/index.js"`, but then we can't specify source and types files. "." is the "alias" for the main export so we gotta life with this.